homeassistant/components/plex/config_flow.py

```diff
@@ -114,6 +133,31 @@ def __init__(self):
             },
         )
 
+    async def async_step_manual_setup(self, user_input=None):
+        """Begin manual configuration."""
+        if len(user_input) > 1:
+            host = user_input.pop(CONF_HOST)
+            port = user_input.pop(CONF_PORT)
+            prefix = "https" if user_input.pop(CONF_SSL) else "http"
+            user_input[CONF_URL] = f"{prefix}://{host}:{port}"
+            return await self.async_step_server_validate(user_input)
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(
+                    CONF_HOST, default=self.discovery_info.get(CONF_HOST)
+                ): str,
+                vol.Required(
+                    CONF_PORT,
+                    default=int(self.discovery_info.get(CONF_PORT, DEFAULT_PORT)),
```

Comment:
It should be. I put this workaround in a while back and I'm not totally sure why. I'll test some more and figure it out.

Reply:
Ah, it was because the discovery payload delivers each value as a string which is fine for the `host`, but not the `port`. When using it as a default value, the validation would fail on submit. I'll move this cast to the discovery step.